### PR TITLE
EIP4844: All public methods take bytes as input (also added explicit G1 validation)

### PR DESCRIPTION
**Pull request by asn-d6**
This is the next step after https://github.com/ethereum/consensus-specs/pull/3219 . With this patch all **Public methods** take bytes as input and do explicit validation (that is, the G1 group validation specified by https://github.com/ethereum/consensus-specs/pull/3223).

We introduce a new validation function called `validate_kzg_g1()` which implements the validation rules for `KZGCommitment` and `KZGProof`. We now do the validation in an explicit manner.

After this patch, `KZGCommitment` and `KZGProof` now represent G1 byte arrays that have been validated. They should only be used by internal functions.

### CODE REVIEW

File: specs/eip4844/polynomial-commitments.md [hunk 1]
```diff
@@ -336,23 +350,29 @@ def blob_to_kzg_commitment(blob: Blob) -> KZGCommitment:
 #### `verify_kzg_proof`
 
 ```python
-def verify_kzg_proof(polynomial_kzg: KZGCommitment,
+def verify_kzg_proof(commitment: Bytes48,
                      z: Bytes32,
                      y: Bytes32,
-                     kzg_proof: KZGProof) -> bool:
+                     kzg_proof: Bytes48) -> bool:
```
Comment (jtraglia) [inline specs/eip4844/polynomial-commitments.md:359]: Nit: I think we should either rename `commitment` to `kzg_commitment` or rename `kzg_proof` to `proof`. My vote goes to renaming `kzg_proof` to `proof` throughout the file.

File: specs/eip4844/polynomial-commitments.md [hunk 2]
```diff
@@ -67,6 +67,8 @@ Functions flagged as "Public method" MUST be provided by the underlying KZG libr
 | - | - | - |
 | `BLS_MODULUS` | `52435875175126190479447740508185965837690552500527637822603658699938581184513` | Scalar field modulus of BLS12-381 |
 | `BYTES_PER_FIELD_ELEMENT` | `uint64(32)` | Bytes used to encode a BLS scalar field element |
+| `G1_POINT_AT_INFINITY` | `Bytes48(b'\xc0' + b'\x00' * 47)` |
+
```
Comment (jtraglia) [inline specs/eip4844/polynomial-commitments.md:72]: This is missing a description/note, also the extra blank line should be removed.

File: specs/eip4844/polynomial-commitments.md [hunk 3]
```diff
@@ -165,6 +168,20 @@ def bytes_to_bls_field(b: Bytes32) -> BLSFieldElement:
...
 
+
+#### `validate_kzg_g1`
+
+```python
+def validate_kzg_g1(b: Bytes48) -> None:
+    """
+    Convert 48-byte value to a G1 point that can be used to hold a KZGCommitment or KZGProof.
+    This performs the validation specified for the types above.
+    """
```
Comment (jtraglia) [inline specs/eip4844/polynomial-commitments.md:179]: Not sure if it's just GitHub or what, but why is "above" wrapped here?

**Review by jtraglia — CHANGES_REQUESTED** [review]
Looking good, I have some minor nits.

File: specs/eip4844/polynomial-commitments.md (hunk 1 above)
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md]: Addressed in af99c1924

File: specs/eip4844/polynomial-commitments.md (hunk 2 above)
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md:72]: Addressed in af99c1924

File: specs/eip4844/polynomial-commitments.md (hunk 3 above)
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md:179]: Doesn't look wrapped to me.

File: specs/eip4844/polynomial-commitments.md [hunk 4]
```diff
@@ -336,35 +353,41 @@ def blob_to_kzg_commitment(blob: Blob) -> KZGCommitment:
...
     """
     Verify KZG proof that ``p(z) == y`` where ``p(z)`` is the polynomial represented by ``polynomial_kzg``.
     Receives inputs as bytes.
     Public method.
     """
-    return verify_kzg_proof_impl(polynomial_kzg, bytes_to_bls_field(z), bytes_to_bls_field(y), kzg_proof)
+    validate_kzg_g1(commitment)
+    validate_kzg_g1(proof)
+
+    return verify_kzg_proof_impl(commitment,
```
Comment (djrtwo) [inline specs/eip4844/polynomial-commitments.md:368]: This needs to be cast explicitly to match the expected types of `_impl`

File: specs/eip4844/polynomial-commitments.md (hunk 3 above)
Comment (jtraglia) [reply specs/eip4844/polynomial-commitments.md:179]: Yeah nevermind. This is fine. Just a bug on GitHub or something.

File: specs/eip4844/polynomial-commitments.md (hunk 4 above)
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md]: Pushed 4204f4c2d1eed07b52af4393c9b20eaeb952e60f which introduces explicit type conversion between "untrusted" bytes and "trusted" BLS types.

File: specs/eip4844/polynomial-commitments.md [hunk 5]
```diff
@@ -449,21 +495,24 @@ def compute_aggregate_kzg_proof(blobs: Sequence[Blob]) -> KZGProof:
 
 ```python
 def verify_aggregate_kzg_proof(blobs: Sequence[Blob],
-                               expected_kzg_commitments: Sequence[KZGCommitment],
-                               kzg_aggregated_proof: KZGProof) -> bool:
+                               expected_kzg_commitments_bytes: Sequence[Bytes48],
+                               kzg_aggregated_proof_bytes: Bytes48) -> bool:
```
Comment (jtraglia) [inline specs/eip4844/polynomial-commitments.md:499]: So these variable names are inconsistent with the other functions. Could we rename these to `expected_commitments` & `aggregated_proof`, respectively. Or if you want to keep the "_bytes" suffix, `expected_commitments_bytes` & `aggregated_proof_bytes`, then add "_bytes" to the other function parameters.

Comment (jtraglia) [inline specs/eip4844/polynomial-commitments.md:499]: For consistency, could we rename these to `expected_commitments_bytes` & `aggregated_proof_bytes`, respectively?

File: specs/eip4844/polynomial-commitments.md [hunk 6]
```diff
@@ -336,35 +379,38 @@ def blob_to_kzg_commitment(blob: Blob) -> KZGCommitment:
 #### `verify_kzg_proof`
 
 ```python
-def verify_kzg_proof(polynomial_kzg: KZGCommitment,
+def verify_kzg_proof(commitment_bytes: Bytes48,
                      z: Bytes32,
                      y: Bytes32,
```
Comment (jtraglia) [inline specs/eip4844/polynomial-commitments.md:384]: Shouldn't `z` and `y` be `z_bytes` and `y_bytes`?

File: specs/eip4844/polynomial-commitments.md (hunk 5 above)
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md]: Done in d6bb1b9ff

File: specs/eip4844/polynomial-commitments.md [hunk 7]
```diff
@@ -336,35 +379,38 @@ def blob_to_kzg_commitment(blob: Blob) -> KZGCommitment:
...
+                                 bytes_to_bls_field(y),
+                                 bytes_to_kzg_proof(proof_bytes))
 ```
 
 
 #### `verify_kzg_proof_impl`
 
 ```python
-def verify_kzg_proof_impl(polynomial_kzg: KZGCommitment,
+def verify_kzg_proof_impl(commitment: KZGCommitment,
```
[the remark this answers is not in the document]
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md:401]: Did the opposite -- as discussed -- in 3db565428

File: specs/eip4844/polynomial-commitments.md (hunk 6 above)
Comment (asn-d6) [reply specs/eip4844/polynomial-commitments.md:384]: We decided to do this rename in the future.

**Review by hwwhww — APPROVED** [review]

**Review by kevaundray — APPROVED** [review]
LGTM

**Pull request merged**